ocs_ci/utility/aws.py

```diff
@@ -862,8 +862,26 @@ def destroy_volumes(cluster_name):
         volumes = aws.get_volumes_by_name_pattern(volume_pattern)
         logger.debug(f"Found volumes: \n {volumes}")
         for volume in volumes:
-            aws.detach_and_delete_volume(
-                aws.ec2_resource.Volume(volume['id'])
-            )
+            # skip root devices for deletion
+            # EBS root device volumes are automatically deleted when
+            # the instance terminates
+            if not check_root_volume(volume):
+                aws.detach_and_delete_volume(
+                    aws.ec2_resource.Volume(volume['id'])
+                )
     except Exception:
         logger.error(traceback.format_exc())
+
+
+def check_root_volume(volume):
+    """
+    Checks whether given EBS volume is root device or not
+
+    Args:
+         volume (dict): EBS volume dictionary
+
+    Returns:
+        bool: True if EBS volume is root device, False otherwise
+
+    """
+    return True if volume['attachments'][0]['DeleteOnTermination'] else False
```

Comment:
I wonder if DeleteOnTermination means it is the root volumes in all cases. Could be that for example, i3 volumes will this set to True as well

Reply:
DeleteOnTermination can change to True even for non-root devices. Currently by default all extra disks created by OCS has option False. 

We need to change the way we are getting volumes, if we want exactly to find the root devices.  Since our teardown is completely broken, I choose the simplest way without any risk.

For i3 instances, if DeleteOnTermination attribute is set to True for extra volumes, then those will be delete automatically once instance is terminated. so we are not leaving any volumes after cluster is destroyed.